pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -66,3 +91,80 @@ func GetProjectVolumeLabels(componentName string) map[string]string {
 	}
 	return volumeLabels
 }
+
+// GetSupervisordVolumeLabels returns the label selectors used to retrieve the supervisord volume
+func GetSupervisordVolumeLabels() map[string]string {
+	image := adaptersCommon.GetBootstrapperImage()
+	_, _, _, imageTag := util.ParseComponentImageName(image)
+
+	supervisordLabels := map[string]string{
+		"name":    adaptersCommon.SupervisordVolumeName,
+		"type":    "supervisord",
+		"version": imageTag,
+	}
+	return supervisordLabels
+}
+
+// CreateAndInitSupervisordVolume creates the supervisord volume and initializes it with odo init
+func CreateAndInitSupervisordVolume(client lclient.Client) (string, error) {
+	supervisordLabels := GetSupervisordVolumeLabels()
+	supervisordVolume, err := client.CreateVolume(supervisordLabels)
+	if err != nil {
+		return "", errors.Wrapf(err, "Unable to create supervisord volume for component")
+	}
+	supervisordVolumeName := supervisordVolume.Name
+
+	err = StartBootstrapSupervisordInitContainer(client, supervisordVolumeName)
+	if err != nil {
+		return "", errors.Wrapf(err, "Unable to start supervisord container for component")
+	}
+
+	return supervisordVolumeName, nil
+}
+
+// StartBootstrapSupervisordInitContainer ...
+func StartBootstrapSupervisordInitContainer(client lclient.Client, supervisordVolumeName string) error {
+	supervisordLabels := GetSupervisordVolumeLabels()
+
+	image := adaptersCommon.GetBootstrapperImage()
+	command := []string{"/usr/bin/cp"}
+	args := []string{
+		"-r",
+		adaptersCommon.OdoInitImageContents,
+		adaptersCommon.SupervisordMountPath,
+	}
+
+	var s *log.Status
+	if log.IsDebug() {
```

Comment:
As an alternative, we can do `if glog.V(4) {` so that it shows up properly when users do `odo push --v <level>` (see https://github.com/openshift/odo/blob/master/pkg/lclient/images.go#L24)

Reply:
Hmm, think will stick with `log.IsDebug()` since it returns true for any `-v` > 0, then its not just bounded by `-v 4`

Reply:
Fair enough, works for me.